app/src/docs/v1.api-spec.yaml

```diff
@@ -169,32 +169,19 @@ paths:
             application/json:
               schema:
                 $ref: "#/components/schemas/Error"
-  "/keyCloak/clients/{keycloakEnv}/{acronym}":
+  "/keycloak/serviceClients":
     get:
-      summary: Returns KeyCloak client details
-      operationId: kcClientGet
+      summary: Returns all registered service clients
+      operationId: kcServiceClientsGet
       tags:
         - KeyCloak Service Client Configuration
-      parameters:
-        - name: keycloakEnv
-          in: path
-          description: "Keycloak realm (dev, test, prod)"
-          required: true
-          schema:
-            type: string
-        - name: acronym
-          in: path
-          description: Name of the application to find the KC client for
-          required: true
-          schema:
-            type: string
       responses:
         "200":
-          description: Returns the Keycloak client details for the acronym specified
+          description: Returns the registered service clients
           content:
             application/json:
               schema:
-                $ref: "#/components/schemas/KcClientDetail"
+                $ref: "#/components/schemas/KcClients"
```

Comment:
I don't think this is correct is it? Not all the fields from this object are coming back from this new API call right? (since it's a simpler one)
Like "commonServiceRoles", does that get included in this api call? It shows in the docs.

Reply:
I think you're correct. my new api call returns the keycloak api response. 
`{
    id: 'd288f328-3d0e-4bd0-8d89-7bdccface260',
    clientId: 'DGRSC_SERVICE_CLIENT',
    name: 'Doc Gen Showcase',
    description: 'Document Generation Showcase Application',
    surrogateAuthRequired: false,
    enabled: true,
    clientAuthenticatorType: 'client-secret',
    redirectUris: [],
    webOrigins: [],
    notBefore: 0,
    bearerOnly: false,
    consentRequired: false,
    standardFlowEnabled: false,
    implicitFlowEnabled: false,
    directAccessGrantsEnabled: false,
    serviceAccountsEnabled: true,
    authorizationServicesEnabled: true,
    publicClient: false,
    frontchannelLogout: false,
    protocol: 'openid-connect',
    attributes: {
      'saml.assertion.signature': 'false',
      'saml.multivalued.roles': 'false',
      'saml.force.post.binding': 'false',
      'saml.encrypt': 'false',
      'saml.server.signature': 'false',
      'saml.server.signature.keyinfo.ext': 'false',
      'exclude.session.state.from.auth.response': 'false',
      saml_force_name_id_format: 'false',
      'saml.client.signature': 'false',
      'tls.client.certificate.bound.access.tokens': 'false',
      'saml.authnstatement': 'false',
      'display.on.consent.screen': 'false',
      'saml.onetimeuse.condition': 'false'
    },
    authenticationFlowBindingOverrides: {},
    fullScopeAllowed: false,
    nodeReRegistrationTimeout: -1,
    protocolMappers: [ [Object], [Object], [Object] ],
    defaultClientScopes: [ 'web-origins', 'roles' ],
    optionalClientScopes: [],
    access: { view: true, configure: true, manage: true }
  },`

i think it's this one: https://www.keycloak.org/docs-api/5.0/rest-api/index.html#_clientrepresentation 
i could refer to that in our api spec. 
or perhaps map the properties we want from the response to a new object and describe that in a new schema.

Reply:
i can fix this on Monday.

Reply:
i've had another go at the api spec. i'm not sure it's quite right but dont want to spend too long on it if possible